app/kuma-ui/pkg/server/server.go

```diff
@@ -10,20 +10,27 @@ import (
 	"github.com/Kong/kuma/pkg/core"
 	core_runtime "github.com/Kong/kuma/pkg/core/runtime"
 	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"strings"
 )
 
 var log = core.Log.WithName("gui-server")
 
 func SetupServer(rt core_runtime.Runtime) error {
-	srv := Server{rt.Config().GuiServer}
+	srv := Server{
+		Config:        rt.Config().GuiServer,
+		ApiServerPort: rt.Config().ApiServer.Port,
+	}
 	if err := core_runtime.Add(rt, &srv); err != nil {
 		return err
 	}
 	return nil
 }
 
 type Server struct {
-	Config *gui_server.GuiServerConfig
+	Config        *gui_server.GuiServerConfig
```

Comment:
Should we deprecate the old `ApiUrl` field ?

Reply:
Added deprecation.